sigoptlite/sigoptlitetest/endpoints/test_best_assignments.py

```diff
@@ -127,108 +143,150 @@ def test_multimetric_thresholds(self, threshold_y1, threshold_y2):
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    self.assert_best_assignments_are_sorted(e, best_assignments_list)
+    assert len(best_assignments_list) > 1
 
-    for best_observation in best_observations_from_logger:
-      for i, metric in enumerate(best_observation["values"]):
+    for best_assignment in best_assignments_list:
+      for i, value in enumerate(best_assignment.values):
         if objectives[i] == "maximize":
-          assert metric["value"] >= thresholds[i]
+          assert value.value >= thresholds[i]
         if objectives[i] == "minimize":
-          assert metric["value"] <= thresholds[i]
-
-    valid_observations = best_assignments_logger.filter_valid_full_cost_observations(observations)
-    expected_valid_observations = [o.get_client_observation(experiment) for o in valid_observations]
-    assert len(best_observations_from_logger) > 1
-    self.assert_observation_lists_are_equal(best_observations_from_logger, expected_valid_observations)
+          assert value.value <= thresholds[i]
 
   def test_multitask(self):
     experiment_meta = self.get_experiment_feature("multitask")
-    experiment = LocalExperimentBuilder(experiment_meta)
-    observations = []
-    for _ in range(20):
-      observation_with_task = self.make_observation(
-        experiment=experiment,
-        assignments=self.make_random_suggestions(experiment)[0].assignments,
-        values=[dict(name="y1", value=numpy.random.rand())],
-        task=numpy.random.choice(experiment_meta["tasks"]),
+    experiment_meta["type"] = "random"
+    e = self.conn.experiments().create(**experiment_meta)
+
+    wrong_optimal_value = 200
+    num_observations = 25
+    for _ in range(num_observations):
+      suggestion = self.conn.experiments(e.id).suggestions().create()
+      if suggestion.task.cost < 1:
+        values = [dict(name="y1", value=wrong_optimal_value)]
+      else:
+        values = [dict(name="y1", value=numpy.random.rand())]
+      self.conn.experiments(e.id).observations().create(
+        suggestion=suggestion.id,
+        values=values,
+        task=suggestion.task,
       )
-      observations.append(observation_with_task)
 
-    best_assignments_logger = BestAssignmentsLogger(experiment)
-    best_observations_from_logger = best_assignments_logger.fetch(observations)
-    assert len(best_observations_from_logger) == 1
-    assert best_observations_from_logger[0]["task"]["cost"] == 1
-    assert best_observations_from_logger[0]["task"]["name"] == "expensive"
+    best_assignments = self.conn.experiments(e.id).best_assignments().fetch()
+    best_assignments_list = list(best_assignments.iterate_pages())
+    assert len(best_assignments_list) == 1
+
+    self.conn.experiments(e.id).observations().fetch()
+    assert best_assignments_list[0].values[0].value != 200
+    assert best_assignments_list[0].values[0].value <= 1
```

Comment:
line 263 implies line 263 is true?

Need to verify that task is the full cost one.

Reply:
the annoying bit is that by design, best assignments does not tell you the task information (same for sigopt server), so not sure how I'll check... let me think about it

Reply:
Ah I see. so we always assume that the best assignments is at full_cost when returned to the user.

Reply:
yea I thought I was losing my mind yesterday for a bit, but I guess that's how it is.

Working on verifying assignments too

Reply:
One idea is to just `to_json()` both objects and compare the `assignments` and `metrics` fields.

Reply:
I think I'm going to keep the test as is, it does its job and verifies that only the full cost tasks are being output by best assignments (even if the explicit task cost isn't being attached)